File: Developer/DeveloperTools/YJLeaks/YJLeaks/YJLeaksViewController.m
//
//  YJLeaksViewController.m
//  YJLeaks
//
//  Created by 阳君 on 2017/1/12.
//  Copyright © 2017年 YJCocoa. All rights reserved.
//

#import "YJLeaksViewController.h"
#import "YJLeaksView.h"

@interface YJLeaksViewController ()

@property (nonatomic, strong) NSTimer *timer; ///<

@end

@implementation YJLeaksViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor greenColor];
//    self.timer = [NSTimer scheduledTimerWithTimeInterval:1 target:self selector:@selector(test) userInfo:nil repeats:true];
    YJLeaksView *view = [[YJLeaksView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    view.backgroundColor = [UIColor whiteColor];
    [self.view addSubview:view];
    view.timer = [NSTimer scheduledTimerWithTimeInterval:1 target:view selector:@selector(test) userInfo:nil repeats:true];
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    [super touchesBegan:touches withEvent:event];
    if (!self.navigationController) {
        [self dismissViewControllerAnimated:YES completion:nil];
    }
}

- (void)test {
    NSLog(@"%@", self);
}

@end

File: Demo/Swift/DeveloperTools/YJLeaks/YJLeaks/YJLeaksViewController.swift
//
//  YJLeaksViewController.swift
//  YJLeaks
//
//  Created by 阳君 on 2019/6/17.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

import UIKit
import YJCocoa
import HandyJSON

var obj: Any?

class YJLeaksViewController: UIViewController & HandyJSON  {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.green
//        self.leakVC()
        self.leakView()
//        self.leakProperty()
        YJLeaks.shared.captureMemoryLeaks(target: self)
        
//        JSONDeserializer<YJLeaksViewController>.deserializeFrom(dict: ["1":"1"])
//        print(YJJSONModel.transformToDict(self))
        
   
    }
    
    func leakVC() {
        obj = self
    }
    
    func leakView() {
        let view = YJLeaksView(frame: CGRect(x: 100, y: 100, width: 100, height: 100))
        view.backgroundColor = UIColor.white
        self.view.addSubview(view)
        obj = view
    }
    
    func leakProperty() {
        
    }
    
    
    
    
}
